sympy/geometry/polygon.py

```diff
@@ -783,6 +783,59 @@ def intersection(self, o):
             return list(ordered(intersection_result))
 
 
+    def cut_section(self, line):
+        """
+        Returns a new polygon (or a segment) formed by the intersecting line
+        and the part of the polygon above it.
+
+        Examples
+        ========
+
+        >>> from sympy import Point, Polygon, Line
+        >>> a, b = 20, 10
+        >>> p1, p2, p3, p4 = [(0, b), (0, 0), (a, 0), (a, b)]
+        >>> rectangle = Polygon(p1, p2, p3, p4)
+        >>> t = rectangle.cut_section(Line((0, 5), slope=0))
+        >>> t
+        Polygon(Point2D(0, 10), Point2D(0, 5), Point2D(20, 5), Point2D(20, 10))
+        >>> t.area
+        100
+        >>> t.centroid
+        Point2D(10, 15/2)
+        """
+        intersection_points = self.intersection(line)
+        if not intersection_points:
+            raise ValueError("This line does not intersect the polygon")
+        points = self.vertices
+        eq = line.equation()
```

Comment:
define x and y as Dummy variables and pass then to equation; there is no point in creating them after the fact below:
```python
x, y = symbols('x y', real=True, cls=Dummy)
eq = line.equation(x, y)
...
```

Reply:
Just a doubt, what is the significance of `_symbols`? I used it as it was used in the creation of the `line.equation()` [here](https://github.com/sympy/sympy/blob/master/sympy/geometry/line.py#L2067)

Reply:
It was a way to handle input from a user for the x and y variables. If they don't provide input then a string is used (e.g. 'x'); if the user provides a symbol then `_symbol` will not create an error. Since your method doesn't need input from the user, you are free to create your own non-clashing symbol using Dummy.